flocker/node/agents/ebs.py

```diff
@@ -309,6 +311,7 @@ def _enable_boto_logging():
 _enable_boto_logging()
 
 
+@with_cmp(["requested", "discovered"])
```

Comment:
> Also, sad that you couldn't use more of the characteristic library, looks like you basically had to roll your own type checking and __repr__ for other reasons :(.

Indeed.  Also sad is that I couldn't use `pyrsistent` here.  That would done the type checking for me, at least (probably not the `__repr__` bit though).  Unfortunately it seems that `PClass` and `Exception` are incompatible base classes so I guess it's not possible to make any exception use `pyrsistent`.